src/transformer/descriptor/helper/helper.ts

```diff
@@ -52,16 +19,34 @@ export namespace TypescriptHelper {
     export function GetDeclarationFromNode(node: ts.Node): ts.Declaration {
         const typeChecker: ts.TypeChecker = TypeChecker();
         const symbol: ts.Symbol = typeChecker.getSymbolAtLocation(node);
-        return symbol.declarations[0];
+        const declaration: ts.Declaration = symbol.declarations[0];
+
+        if (ts.isImportSpecifier(declaration)) {
+            return GetDeclarationForImport(declaration) as ts.Declaration;
+        }
+
+        return declaration;
     }
 
-    export function GetDeclarationForImport(node: ts.ImportClause | ts.ImportSpecifier): ts.Node {
+    export function GetDeclarationForImport(node: ts.ImportClause | ts.ImportSpecifier): ts.TypeNode | ts.Declaration {
         const typeChecker: ts.TypeChecker = TypeChecker();
         const symbol: ts.Symbol = typeChecker.getSymbolAtLocation(node.name);
         const declaredType: ts.Type = typeChecker.getDeclaredTypeOfSymbol(symbol);
         return GetDeclarationFromType(declaredType);
     }
 
+    export function GetParameterOfNode(node: ts.EntityName): ts.NodeArray<ts.TypeParameterDeclaration> {
+        const declaration: ts.Declaration = GetDeclarationFromNode(node);
+
+        if (ts.isImportSpecifier(declaration)) {
```

Comment:
Since now the check `isImportSpecifier` is done in GetDeclarationFromNode is this still required here?

Reply:
yes you are right